packages/backend/src/execution/executors/QueryProviderExecutor.ts

```diff
@@ -1,22 +1,73 @@
 import { QueryProvider } from '../../QueryProvider';
 import { AnyQuery } from '../../types';
-import { QueryNode } from '../../query/createQueryTree';
 import { RefContext, createRefContext } from '../../refs/RefContext';
 import { QueryExecutor } from '../types';
 import { ColumnRef } from '../../refs/ColumnRef';
+import { Table, Where } from '@synthql/queries';
 
-export class QueryProviderExecutor implements QueryExecutor {
-    private providersByTable: Map<string, QueryProvider>;
-    constructor(providers: QueryProvider[]) {
+export class QueryProviderExecutor<DB> implements QueryExecutor {
+    private providersByTable: Map<string, QueryProvider<DB, Table<DB>>>;
+
+    constructor(providers: QueryProvider<DB, Table<DB>>[]) {
         this.providersByTable = new Map(providers.map((p) => [p.table, p]));
     }
 
+    extract(where: Where<DB, Table<DB>>) {
```

Comment:
I see you did some "divide and conquer" by splitting the function into it's own 👍 great call.

A few improvements we can make:
1. `extract` is not a great name. let's give it a much more explicit name. How about `convertWhereToQueryProviderInput`. It's not sexy, but it's not an API name, so it's totally fine to make it overly verbose (it literally has just 1 usage)
2. Move it outsude of the QueryProviderExecutor so you can actually write unit tests for it. This function is completely stateless, so making it actually stateless (i.e. just a plain old function) let's future maintainers quickly understand that the output only depends on the input.